### PR TITLE
New Trait: Oni Damage Bonuses

### PR DESCRIPTION
**Pull request by angelofallars**
# Description

Adds three new 1-point traits for Onis that **redistribute** your damage bonuses, allowing you to specialize in Slash or Piercing instead of Blunt, or be a generalist who is equally effective with all melee Brute weapons. The three new traits are **Swashbuckler**, **Spearmaster**, and **Weapons Generalist**.

| Traits / Damage Bonus  | Blunt | Slash | Piercing |
|------------------------|-------|-------|----------|
| Base Oni (No Traits)   | 35%   | 20%   | 20%      |
| **Swashbuckler**       | 20%   | 35%   | 20%      |
| **Spearmaster**        | 20%   | 20%   | 35%      |
| **Weapons Generalist** | 25%   | 25%   | 25%      |

# Changelog

:cl: Skubman
- add: Add three new 1-point traits for Onis that allow you to specialize in Slash or Piercing damage or be a melee weapons generalist.

### CODE REVIEW

File: Resources/Prototypes/Traits/categories.yml
Comment (DangerRevolution) [inline Resources/Prototypes/Traits/categories.yml:1]: No new categories until we get scrolling; stick this in physical

File: Resources/Locale/en-US/traits/categories.ftl
Comment (DangerRevolution) [inline Resources/Locale/en-US/traits/categories.ftl:1]: No new categories until we get scrolling; stick this in physical

File: Content.Server/Traits/Assorted/OniDamageModifierComponent.cs
Comment (DangerRevolution) [inline Content.Server/Traits/Assorted/OniDamageModifierComponent.cs:1]: IMO, this should be (one day) a more generic system that can give any species these modifiers; obviously changing the oni modifier to be more generic. Unrelated to this PR though, just giving thoughts for sauce.

**Review by DangerRevolution — CHANGES_REQUESTED** [review]

File: Content.Server/Traits/Assorted/OniDamageModifierComponent.cs
Comment (angelofallars) [reply Content.Server/Traits/Assorted/OniDamageModifierComponent.cs:1]: We just need to extract the melee stuff from the OniSystem and it'll be generic

File: Resources/Prototypes/Traits/categories.yml
Comment (angelofallars) [reply Resources/Prototypes/Traits/categories.yml:1]: Resolved with 75c01358206722e8895b3127ada4dee5929040ef

File: Resources/Locale/en-US/traits/categories.ftl
Comment (angelofallars) [reply Resources/Locale/en-US/traits/categories.ftl:1]: Resolved with 75c01358206722e8895b3127ada4dee5929040ef

**Review by DangerRevolution — APPROVED** [review]

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**